src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -69,13 +69,13 @@ function HeatmapVis(props: Props): JSX.Element {
         abscissaConfig={{
           domain: abscissaDomain,
           showGrid,
-          isIndexAxis: !abscissas,
+          isIndexAxis: !abscissaParams.values,
```

Comment:
`abscissas` is never `undefined` as it is set to `range(dataArray.size)` when destructuring `abscissaParams.values`.

Changing the condition to `abscissaParams.values` restores the check validity.

Reply:
Hmm yeah I get you. You could store this condition in a constant at the top of the component and extract the `range` call out of `getPixelEdges`. Is this what you initially had in mind as a refactoring?

```ts
const isAbscissaIndexAxis = !abscissaParams.values;
const abscissas = isAbscissaIndexAxis ? range(cols + 1) : getPixelEdges(abscissaParams.values, cols);
```

It's still not great, is it? We could also pass `isIndexAxis` as part of each axis' params object prop, but this would complexify the public API...

I like your current solution the best for now. Unless you had something else in mind?

Reply:
> Hmm yeah I get you. You could store this condition in a constant at the top of the component and extract the `range` call out of `getPixelEdges`. Is this what you initially had in mind as a refactoring?
> 
> ```ts
> const isAbscissaIndexAxis = !abscissaParams.values;
> const abscissas = isAbscissaIndexAxis ? range(cols + 1) : getPixelEdges(abscissaParams.values, cols);
> ```
> 
> It's still not great, is it? 

I followed the exact same line of thought ! I feel we can do better but did not manage to find a satisfying solution on my own.

> We could also pass `isIndexAxis` as part of each axis' params object prop, but this would complexify the public API...
> 
> I like your current solution the best for now. Unless you had something else in mind?

I tried a few things here and there but in the end, I think this fix is the most straightforward solution